### PR TITLE
Always output tx

### PR DESCRIPTION
**Pull request by asonnino**
Tiny PR ensuring consensus always returns the transaction to the end user after processing (even if it encountered an error).

### CODE REVIEW

**Review by huitseeker — COMMENTED** [review]
Now this looks good to me from the `SerializedTransactionDigest` side, but since you renamed `client -> executor` (which I think is a good move), I'm gonna need at least a paragraph doc comment in `executor/lib.rs` (or possibly a Readme) that explains what execution is (and isn't) in this code base and what justifies the name, written so that somebody used to working with a classic blockchain architecture (Eth, Celo, Tendermint ...) can understand.

File: executor/src/core.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ pub struct Core<State: ExecutionState, PublicKey: VerifyingKey> {
     /// Receive ordered consensus output to execute.
     rx_subscriber: Receiver<ConsensusOutput<PublicKey>>,
     /// Outputs executed transactions.
-    tx_output: Sender<SubscriberResult<SerializedTransactionDigest>>,
+    tx_output: Sender<(SubscriberResult<()>, SerializedTransactionDigest)>,
```
Comment (asonnino) [inline executor/src/core.rs:41]: Revering this for the moment but will fix it later

Comment (asonnino) [reply executor/src/core.rs:41]: Made an issue here: #1489 (it needs to be fixed along with Sui)

**Review by laura-makdah — APPROVED** [review]
I'm glad to have one less thing named "Client".

**Pull request merged**